src/normalize/text.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Editor, NodeEntry, Text, Transforms } from 'slate'
+
+import { Caption, Figure, Foreign, Quotation, Term, Title } from '../interfaces'
+
+/**
+ * Normalize text nodes
+ *
+ * Return true if entry was normalized and false otherwise.
+ */
+export default function normalizeText(editor: Editor, entry: NodeEntry): boolean {
+    const [node, path] = entry
+
+    if (Caption.isCaption(node)) {
+        // Caption is only allowed as the last child of a figure.
+        const [parent] = Editor.parent(editor, path)
+        if (!Figure.isFigure(parent)) {
+            Transforms.setNodes(editor, { type: 'paragraph' }, { at: path })
+            return true
+        }
+    }
+
+    if (Title.isTitle(node)) {
+        // Title must be the first child of it's parent.
+        if (path[path.length - 1] > 0) {
+            Transforms.setNodes(editor, { type: 'paragraph' }, { at: path })
+            return true
+        }
+    }
+
+    if (Quotation.isQuotation(node)) {
+        // A quotation must not be empty.
+        if (node.children.length === 0) {
+            Transforms.removeNodes(editor, { at: path })
+            return true
+        }
+
+        // A quotation must have other elements than just a title.
+        if (node.children.length === 1 && Title.isTitle(node.children[0])) {
+            Transforms.insertNodes(editor, {
+                type: 'paragraph',
+                children: [],
+            }, { at: [...path, 1] })
+            return true
+        }
+
+        // A quotation must not consist only of another quotation.
```

Comment:
I addressed this in one of the tests, but I'm also leaving a comment here
If we don't allow creating nested quotation like this then creating a structure like this:
```
<q>
  <q>
    asd
  </q>
  <p>asdsad</p>
</q>
```
would be unintuitive because suer would have to start with creating two paragraphs inside quote and then first one with a quote